### PR TITLE
DDSSQLFilter Behavioral tests [13789]

### PR DESCRIPTION
**Pull request by MiguelCompany**
This PR adds tests for DDSSQLFilter module. Still pending:

* [x] Tests checking data promotion
* [x] Tests for compound expressions
* [x] Tests for parameters update

### CODE REVIEW

**Review by JLBuenoLopez-eProsima — COMMENTED** [review]
Only a few suggestions. In general it looks really good!

File: test/unittest/dds/topic/DDSSQLFilter/DDSSQLFilterTests.cpp [hunk 1]
```diff
@@ -420,6 +420,78 @@ TEST_F(DDSSQLFilterTests, type_compatibility_compare)
     }
 }
 
+/**
+ * Singleton that holds the serialized payloads to be evaluated
+ */
+struct DDSSQLFilterValueGlobalData
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply test/unittest/dds/topic/DDSSQLFilter/DDSSQLFilterTests.cpp:519]: You are right, should be a class.

File: test/unittest/dds/topic/DDSSQLFilter/DDSSQLFilterTests.cpp [hunk 2]
```diff
@@ -1234,6 +1235,70 @@ static std::vector<DDSSQLFilterValueParams> get_test_filtered_value_string_input
...
+    results[4][1] = { false, false, true, true, true };
+    results[4][2] = { false, false, false, false, false };
+    results[4][3] = { false, false, false, false, false };
+    results[4][4] = { false, false, false, false, false };
+    // GE
+    results[5][0] = { true, true, true, true, true };
+    results[5][1] = { true, true, true, true, true };
+    results[5][2] = { false, false, true, true, true };
+    results[5][3] = { false, false, true, true, true };
+    results[5][4] = { false, false, true, true, true };
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply test/unittest/dds/topic/DDSSQLFilter/DDSSQLFilterTests.cpp:1556]: They all run in less than 10 seconds, I think we can afford the penalty.

**Comment by MiguelCompany** [comment]
@JLBuenoLopez-eProsima I think I have addressed all your comments.

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Pull request merged**